.github/workflows/app.yml

```diff
@@ -31,25 +31,25 @@ jobs:
         uses: actions/checkout@v4
         with:
           persist-credentials: false
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 10
       - name: Install node
         uses: actions/setup-node@v4
         with:
           node-version: "22.x"
-          cache: npm
-          cache-dependency-path: app/package-lock.json
-      - name: Install npm dependencies
+          cache: pnpm
+          cache-dependency-path: app/pnpm-lock.yaml
+      - name: Install node dependencies
         working-directory: app
-        run: |
-          npm ci
+        run: pnpm install
```

Comment:
For posterity, `pnpm install` automatically runs with `--frozen-lockfile` (equivalent of `npm ci`) when it detects its in a CI environment. Great!